framework/doc/content/source/csg/CSGBase.md

```diff
@@ -0,0 +1,663 @@
+std::string new_join_name = input_mg_names[1] + "_univ";
+std::string new_base_name = input_mg_names[0] + "_univ";
+// joining via this method will move both roots into new universes with new names
+csg_obj->joinOtherBase(inp_csg_obj, new_base_name, new_join_name);
+```
+
+For all of these join methods, any non-root universes will remain unchanged and simply added to the list of universes for the current `CSGBase` object.
+Similarly, all incoming cells and surfaces are added alongside existing cells and surfaces.
+
+!alert! note title=Object Naming Uniqueness
+
+It is very important when using the `joinOtherBase` method that all `CSGSurfaces`, `CSGCells`, and `CSGSurfaces` are uniquely named so that errors are not encountered when combining sets of objects.
+An error will be produced during the join process if an object of the same type and name already exists.
+See [recommendations for naming](#object-naming-recommendations) below.
+
+!alert-end!
+
+## Accessing CSG-related Methods
+
+All [!ac](CSG) methods related to creating or changing a [!ac](CSG) object must be called through `CSGBase`.
+Calls that retrieve information only but do not manipulate an object (such as `getName` methods) can be called on the object directly.
+For example, if a cell were to be created, the current name and region could be retrieved directly from the `CSGCell` object, but if the name or region needed to be changed, that would need to be handled through `CSGBase`:
+
+```cpp
+// a cell is created using CSGBase
+auto cell = csg_obj->createCell("cell_name", region);
+// the current name and region of that cell can be retrieved directly from the CSGCell object
+auto name = cell->getName();
+auto region = cell->getRegion();
+// changing the name and region requires using methods in CSGBase
+csg_obj->renameCell(cell, "new_name");
+csg_obj->updateCellRegion(cell, new_region);
+```
+
+This ensures proper accounting of all [!ac](CSG)-related objects in the `CSGBase` instance.
+Consult the Doxygen documentation for information on all object-specific methods.
+
+## Object Naming Recommendations
+
+For each new [!ac](CSG) element (`CSGSurface`, `CSGCell`, and `CSGUniverse`) that is created, a unique name identifier (of type `std::string`) must be provided (`name` parameter for all creation methods).
+A recommended best practice is to include the mesh generator name (which can be accessed with `this->getName()` in any MeshGenerator class) as a part of that object name.
+This `name` is used as the unique identifier within the `CSGBase` instance.
+Methods for renaming objects are available as described in the above sections to help prevent issues and errors.
+
+## Example Implementation
+
+Provided here is an example implementation of the `generateCSG` method for a simple example [source/meshgenerators/MeshGenerator.md] that creates an infinite rectangular prism given an input parameter for `side_length`.
+The code snippets provided here correspond to the `.C` file.
+
+```cpp
```

Comment:
let's just add this to MOOSE as an example CSGGenerator

Reply:
I think I ll backtrack here. Let's just mention that these examples are only runnable with the test executable.

I don't really want more MeshGenerators doing simple things like a cylinder, and only doing the CSG. 
So while these go as test objects, the existing simple MGs will become better examples as they get generateCSG() implemented

Reply:
noted - yes I think once we implement this in a few existing MGs we can update this to point those and then add the corresponding input/output to the examples folder.

Reply:
okay never mind, I see you want the test file referenced